packages/contracts/test/plugin/plugin-setup-processor.ts

```diff
@@ -571,120 +688,123 @@ describe('Plugin Setup Processor', function () {
           psp.applyUninstallation(
             targetDao.address,
-            plugin,
-            pluginSetup,
-            pluginSetupMockRepoAddress,
+            proxy,
+            setupV1.address,
+            repoUUPS.address,
             [],
             []
           )
         ).to.be.revertedWith(customError('HelpersHashMismatch'));
       });
 
       it('revert bad permissions is passed', async () => {
-        const pluginSetup = pluginSetupV1Mock.address;
-
-        const {plugin, helpers, prepareInstallpermissions} =
-          await prepareInstallation(
-            psp,
-            targetDao.address,
-            pluginSetup,
-            pluginSetupMockRepoAddress,
-            EMPTY_DATA
-          );
-
-        await psp.applyInstallation(
-          targetDao.address,
-          pluginSetup,
-          pluginSetupMockRepoAddress,
-          plugin,
-          prepareInstallpermissions
-        );
-
         await psp.callStatic.prepareUninstallation(
           targetDao.address,
-          plugin,
-          pluginSetup,
-          pluginSetupMockRepoAddress,
-          helpers,
+          proxy,
+          setupV1.address,
+          repoUUPS.address,
+          helpersV1,
           EMPTY_DATA
         );
 
+        let badPermissions: PermissionOperation[] = [];
         await expect(
           psp.applyUninstallation(
             targetDao.address,
-            plugin,
-            pluginSetup,
-            pluginSetupMockRepoAddress,
-            helpers,
-            []
+            proxy,
+            setupV1.address,
+            repoUUPS.address,
+            helpersV1,
+            badPermissions
           )
         ).to.be.revertedWith(customError('PermissionsHashMismatch'));
       });
 
-      it('correctly complete an uninstallation process', async () => {
-        const pluginSetup = pluginSetupV1Mock.address;
-
-        const {plugin, helpers, prepareInstallpermissions} =
-          await prepareInstallation(
-            psp,
-            targetDao.address,
-            pluginSetup,
-            pluginSetupMockRepoAddress,
-            EMPTY_DATA
-          );
-
-        await psp.applyInstallation(
-          targetDao.address,
-          pluginSetup,
-          pluginSetupMockRepoAddress,
-          plugin,
-          prepareInstallpermissions
-        );
-
-        const tx = await psp.prepareUninstallation(
-          targetDao.address,
-          plugin,
-          pluginSetup,
-          pluginSetupMockRepoAddress,
-          helpers,
-          EMPTY_DATA
-        );
+      it('applies a prepared uninstallation', async () => {
+        await expect(
+          uninstallHelper(psp, targetDao, proxy, helpersV1, setupV1, repoUUPS)
+        ).to.not.be.reverted;
+      });
 
-        const event = await findEvent(tx, 'UninstallationPrepared');
-        const {permissions} = event.args;
+      it.skip('applies multiple prepared uninstallations of the same plugin', async () => {
```

Comment:
why this test is skipped?
I thought it is a good thing that we prevent re-granting ,re-revoking...etc this way the permission manager will not be polluted.
is there plans to allow re-granting and re-revoking...etc?